[doc] Update i2c documentation. (#16697)
lowRISC/opentitan: hw/ip/i2c/doc/_index.md

    - 7-bit target address
- Support for the following optional capabilities:
    - Clock stretching in the host mode

>     - Automatic clock stretching in the target mode<sup>2</sup>

Reviewer: Nit: should this bullet point end in a period?

Author: ah for this one i just followed the previous bullets, didn't want to make this one stand out.

- SW may reset I2C block using the Reset Manager.

<sup>1</sup> lowRISC is avoiding the fraught terms master/slave and defaulting to host/target where applicable.

> <sup>2</sup> The target is only compatible with hosts that support clock stretching. For hosts that do not support clock stretching, it is expected that there must be an additional protocol to guarantee there is always sufficient space and data.  These protocols are not in scope of this document.

Reviewer: Nit: each sentence should have its own line.

![](I2C_acq_fifo_write.svg)

If the transaction is a read operation (R/W bit = 1), the target pulls bytes out of TX FIFO and transmits them to the bus until the host signals the end of the transfer by sending a NACK signal.

> If TX FIFO holds no data, or if the ACQ FIFO contains more than 1 entry, and thus a potentially unhandled command from the host, the target will hold SCL low to stretch the clock and give software time to write data bytes into TX FIFO or handle the available command.

Reviewer: ```suggestion
If TX FIFO holds no data or if the ACQ FIFO contains more than 1 entry the target will hold SCL low to stretch the clock and give software on the host time to write data bytes into TX FIFO or handle the available command.
```
Nit: maybe removing the nested clause makes the sentence less confusing?

    - If there is no space in the ACQ FIFO to receive such a write, the target stretches the clock after the ACK bit and waits for software to make space.
    - The `acq_full` interrupt is generated to alert software to such a situation.

#### Stretching during write
    - Similar to the scenario above, if the host tries to write a data byte into the ACQ FIFO when there is no available space, the clock is also stretched after the ACK bit.
    - The `acq_full` interrupt is generated to alert software to such a situation.

#### Stretching during read
    - When a target device receives a start and read command, it may stretch the clock for either of the following two reasons.
      - If there is no data available to be sent back (TX FIFO empty case), the target stretches the clock until data is made available by software.
      - If there is more than 1 entry in the ACQ FIFO.
        - Having more than 1 entry in the ACQ FIFO suggests there is potentially an unhandled condition (STOP / RESTART) or an unhandled command (START) that requires software intervention before the read can proceed.
    - The `tx_stretch` interrupt is generated to alert software to such a situation.

> In all cases described above, a target begins to stretch the clock after the ACK bit.

Reviewer 2: This is going to look like it's under the "Stretching during read" label, isn't it? If so, maybe this text should be adjusted to go above the list, or the labels should become part of the list instead?

Author: o yeah good point.
